gnomad_hail/utils/relatedness.py

```diff
@@ -356,37 +395,41 @@ def discard_multi_parents_children(trios: List[hl.Trio]):
 
         return discard_multi_parents_children(trios)
 
+    # Get all the relations we care about:
+    # => Remove unrelateds and duplicates
+    dups = duplicate_samples_ht.aggregate(
+        hl.agg.explode(
+            lambda dup: hl.agg.collect_as_set(dup),
+            duplicate_samples_ht.filtered
+        ),
+        _localize=False
+    )
+    relationship_ht = relationship_ht.filter(
+        ~dups.contains(relationship_ht[i_col]) &
+        ~dups.contains(relationship_ht[j_col]) &
+        (relationship_ht[relationship_col] != UNRELATED)
+    )
+
     # Check relatedness table format
     if not relationship_ht[i_col].dtype == relationship_ht[j_col].dtype:
         logger.error("i_col and j_col of the relatedness table need to be of the same type.")
 
-    # If `relatedness_ht` is keyed by a Struct, attempt to extract `s` instead
-    if isinstance(relationship_ht[i_col], hl.expr.StructExpression):
-        if 's' in relationship_ht[i_col]:
-            logger.warn(f"Pedigrees can only be constructed from string IDs, but your relatedness_ht ID column is of type: {relationship_ht[i_col].dtype}. Attempting to use relatedness_ht[{i_col}].s")
-            relationship_ht = relationship_ht.key_by(  # Needed for familial inference at this point -- should be generalized
+    # If i_col and j_col aren't str, then convert them
+    if not isinstance(relationship_ht[i_col], hl.expr.StringExpression):
+        logger.warning(f"Pedigrees can only be constructed from string IDs, but your relatedness_ht ID column is of type: {relationship_ht[i_col].dtype}. Expression will be converted to string in Pedigrees.")
+        if isinstance(relationship_ht[i_col], hl.expr.StructExpression):
```

Comment:
are there cases where this might not be a string but also not a struct and what would happen to those?

Reply:
good point, I'll raise an error